### PR TITLE
Added transfer subscription tests

### PR DESCRIPTION
**Pull request by MakisChristou**
PR addresses [this](https://github.com/orgs/vechain/projects/22/views/1?filterQuery=e2e&pane=issue&itemId=72240898) issue. For the negative path I added an optional `errorCallBack` to the `thorClient`. Also to test for a valid sender I added the option to inject the wallet to the `fundAccount` function but let me know if you have better ideas on how to do this.

### CODE REVIEW

File: contracts/SimpleTransfer.sol [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.20;
+
+contract SimpleTransfer {
+    event TransferInitiated(address indexed txOrigin, address indexed sender, uint256 amount);
+
+    function transfer(address payable recipient, uint256 amount) external payable {
+        require(msg.value == amount, "Incorrect amount sent");
```
Comment (darrenvechain) [inline contracts/SimpleTransfer.sol:8]: Do we need the amount parameter?

We could just `require(msg.value > 0, ....)`

File: src/thor-client.ts [hunk 2]
```diff
@@ -469,13 +470,30 @@ class ThorClient {
...
+        ws.onerror = (event: any) => {
+            if (errorCallback) {
+                errorCallback(event)
+            }
+            const error = new Error(`WebSocket error: ${event.message}`)
+            console.error(error)
+        }
+
+        ws.onclose = (event: any) => {
+            const error = new Error(`WebSocket onclone: ${event.message}`)
```
Comment (darrenvechain) [inline src/thor-client.ts:493]:
```suggestion
            const error = new Error(`WebSocket onclose: ${event.message}`)
```

File: src/thor-client.ts [hunk 3]
```diff
@@ -469,13 +470,30 @@ class ThorClient {
...
             const data = JSON.parse(event.data)
             callback(data)
         }
 
+        ws.onerror = (event: any) => {
+            if (errorCallback) {
+                errorCallback(event)
+            }
+            const error = new Error(`WebSocket error: ${event.message}`)
+            console.error(error)
```
Comment (darrenvechain) [inline src/thor-client.ts:489]: I think we can exclude console logs here

File: src/thor-client.ts [hunk 4]
```diff
@@ -469,13 +470,30 @@ class ThorClient {
...
+        }
+
+        ws.onclose = (event: any) => {
+            const error = new Error(`WebSocket onclone: ${event.message}`)
+            console.error(error)
```
Comment (darrenvechain) [inline src/thor-client.ts:494]: same here, no need for logs, should be captured in callback

File: test/thorest-api/subscriptions/ws-transfers.test.ts [hunk 5]
```diff
@@ -1,41 +1,241 @@
...
+        Client.raw.subscribeToTxPool((txId) => {
+            txpoolTxs.push(txId)
+        })
+
+        // Sleep for 1 sec to ensure the beat is received
+        await new Promise((resolve) => setTimeout(resolve, 1000))
+
+        expect(transferTxs).not.toBeUndefined()
+        expect(txpoolTxs).not.toBeUndefined()
+        expect(transferTxs).toEqual(txpoolTxs)
```
Comment (darrenvechain) [inline test/thorest-api/subscriptions/ws-transfers.test.ts:200]: @MakisChristou looks like this test was retried twice, pointing to instability. (see below)

FYI @leszek-vechain

**Review by leszek-vechain — APPROVED** [review]

**Pull request merged**